### PR TITLE
Solve race condition in qstat_proxy.sh

### LINKED ISSUE #5168: Solve possible race condition in qstat_proxy.sh
There is potentially a race condition in the between the two if-statements in:

https://github.com/equinor/ert/blob/ce2f84f4c4d7263a7b8aadaa3942db9df49ac12c/src/ert/_c_wrappers/job_queue/qstat_proxy.sh#L92-L104

where the `$proxyfile` has zero content in the second `if`-sentence while it had non-zero content in the first. This might explain errors seen in https://github.com/equinor/scout/issues/791

### PR DESCRIPTION
**Pull request by berland**
The code read the proxyfile from disk multiple times. Other processes can start writing to the file during the duration of this script, which can explain observations where this script gives zero exit code and zero content.

**Issue**
Resolves #5168 

**Approach**

Read the contents into memory instead of repeated reads from disk.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/job_queue/qstat_proxy.sh [hunk 1]
```diff
@@ -89,22 +89,26 @@ if [ $proxyage_seconds -gt $CACHE_INVALID ]; then
...
 
+# Read the contents into memory to avoid race conditions. Other
+# instances of this proxy might edit the proxyfile at any time.
+proxyfilecontents=$(cat $proxyfile)
+
 # Replicate qstat's error behaviour:
 if [ -n "$1" ]; then
-    grep -e "Job Id: ${1}" $proxyfile >/dev/null 2>&1 || {
-        echo "qstat: Unknown Job Id $1" && cat $proxyfile >&2 && exit 1;
+    echo "$proxyfilecontents" | grep -e "Job Id: ${1}" >/dev/null 2>&1 || {
```
Comment (hnformentin) [inline src/ert/_c_wrappers/job_queue/qstat_proxy.sh:98]: Why "" is needed?

**Comment by hnformentin** [comment]
Could you describe if/how you have tested the behaviour?

File: src/ert/_c_wrappers/job_queue/qstat_proxy.sh (hunk 1 above)
Comment (berland) [reply src/ert/_c_wrappers/job_queue/qstat_proxy.sh:98]: https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Word-Splitting

**Comment by hnformentin** [comment]
Test new parser is failing. Do you know if this is related to the PR?

**Comment by berland** [comment]
> Test new parser is failing. Do you know if this is related to the PR?

That must be unrelated.

**Comment by berland** [comment]
> Could you describe if/how you have tested the behaviour? Explained in standup that this is requires repetition of running the job.

The qstat shell script is already extensively tested, and the change passes all those. 

The problem solved by this PR is only observed in full-scale integration tests, such as the hm-tutorial in komodo-releases.

**Comment by berland** [comment]
This will not fly due to #5179

**Comment by berland** [comment]
This PR was replaced by #5182

**Pull request closed without merging**